### PR TITLE
fix: some icons didn't follow icon theme

### PR DESCRIPTION
**Pull request by rodolforg**
Reported-by: pablogil
Via: https://forums.synfig.org/t/modern-iconset-light-and-dark/13560/31

### CODE REVIEW

File: synfig-studio/src/gui/instance.cpp [hunk 1]
```diff
@@ -1293,10 +1297,8 @@ Instance::make_param_menu(Gtk::Menu *menu,synfig::Canvas::Handle canvas, synfiga
 			}
 		}
 
-		item = Gtk::manage(new Gtk::ImageMenuItem(
-			*manage(new Gtk::Image(
-				Gtk::StockID("gtk-convert"),
-				Gtk::ICON_SIZE_MENU )),
+        item = Gtk::manage(new Gtk::ImageMenuItem(
```
Comment (ice0) [inline synfig-studio/src/gui/instance.cpp:1300]: spaces

File: synfig-studio/src/gui/instance.cpp [hunk 2]
```diff
@@ -1114,7 +1118,7 @@ Instance::add_actions_to_menu(Gtk::Menu *menu, const synfigapp::Action::ParamLis
 		if(!(iter->category&synfigapp::Action::CATEGORY_HIDDEN))
 		{
 			Gtk::MenuItem *item = Gtk::manage(new Gtk::ImageMenuItem(
-				*Gtk::manage(new Gtk::Image(get_action_stock_id(*iter),Gtk::ICON_SIZE_MENU)),
+                *Gtk::manage(create_image_from_icon(get_action_icon_name(*iter), Gtk::ICON_SIZE_MENU)),
```
Comment (ice0) [inline synfig-studio/src/gui/instance.cpp:1121]: spaces

File: synfig-studio/src/gui/instance.cpp [hunk 3]
```diff
@@ -1094,7 +1098,7 @@ Instance::add_actions_to_menu(Gtk::Menu *menu, const synfigapp::Action::ParamLis
...
```
Comment (ice0) [inline synfig-studio/src/gui/instance.cpp:1101]: spaces

File: synfig-studio/src/gui/instance.cpp [hunk 4]
```diff
@@ -1047,8 +1050,9 @@ Instance::add_actions_to_menu(Gtk::Menu *menu, const synfigapp::Action::ParamLis
 				continue;
 
 			Gtk::MenuItem *item = Gtk::manage(new Gtk::ImageMenuItem(
-				*Gtk::manage(new Gtk::Image(get_action_stock_id(*iter),Gtk::ICON_SIZE_MENU)),
+                *Gtk::manage(create_image_from_icon(get_action_icon_name(*iter), Gtk::ICON_SIZE_MENU)),
```
Comment (ice0) [inline synfig-studio/src/gui/instance.cpp:1053]: spaces

File: synfig-studio/src/gui/instance.cpp [hunk 5]
```diff
@@ -95,6 +95,9 @@ int studio::Instance::instance_count_=0;
 static Gtk::Image*
 create_image_from_icon(const std::string& icon_name, Gtk::IconSize icon_size)
 {
+    if (icon_name == "image-missing")
```
Comment (ice0) [inline synfig-studio/src/gui/instance.cpp:98]: spaces

**Review by ice0 — APPROVED** [review]

**Comment by ice0** [comment]
Merged. Thank you!

**Comment by pgilfernandez** [comment]
I finally had the time to try the fix: it works like a charm =D

Thanks

File: synfig-studio/src/gui/iconcontroller.cpp [hunk 6]
```diff
@@ -446,29 +446,31 @@ studio::get_tree_pixbuf(Type &type)
...
-	else if(action.task=="set_off")		stock_id=Gtk::Stock::NO;
-	else if(action.task=="set_on")		stock_id=Gtk::Stock::YES;
-	else								stock_id=Gtk::StockID("synfig-"+
-															  action.task);
-	return stock_id;
+	// maps action task -> icon name
+	const std::map<std::string, std::string> action_icon_map = {
+		{"add",         "list-add"},
+		{"insert",      "list-add"},
+		{"remove",      "gtk-delete"},
```
Comment (pgilfernandez) [inline synfig-studio/src/gui/iconcontroller.cpp:456]: this line should be "edit-delete" or the icon will be messed up...

**Comment by pgilfernandez** [comment]
As I reported in https://github.com/synfig/synfig/issues/3116 (maybe too soonish as the PR hasn't been merged yet...) I found that some "delete" icons are gone. I had time to test it by myself and found out that the problem was some references changed to "gtk-delete" while the correct one should be "edit-delete" as it's the only one that seems to be used in Synfig.
If you revert all "gtk-delete" references in:

- dock_layers.cpp
- dock_metadata.cpp
- dialog_workspaces.glade

But keep "edit-delete" in `iconcontroller.cpp` as this way it works back again ;)

**Pull request merged**